<a href="https://colab.research.google.com/github/axefx/DS-Unit-1-Sprint-2-Statistics/blob/master/module1/Axel_Corro_LS_DS13_121_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [16]:
### YOUR CODE STARTS HERE
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

--2020-02-17 21:08:17--  https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18171 (18K) [application/x-httpd-php]
Saving to: ‘house-votes-84.data.1’

house-votes-84.data 100%[===================>]  17.75K  --.-KB/s    in 0.06s   

2020-02-17 21:08:18 (286 KB/s) - ‘house-votes-84.data.1’ saved [18171/18171]



In [17]:
import pandas as pd

df = pd.read_csv('house-votes-84.data', names=['party','handicapped-infants','water-project',
                          'budget','physician-fee-freeze', 'el-salvador-aid',
                          'religious-groups','anti-satellite-ban',
                          'aid-to-contras','mx-missile','immigration',
                          'synfuels', 'education', 'right-to-sue','crime','duty-free',
                          'south-africa'])
df.head()

party handicapped-infants water-project  ... crime duty-free south-africa
0  republican                   n             y  ...     y         n            y
1  republican                   n             y  ...     y         n            ?
2    democrat                   ?             y  ...     y         n            n
3    democrat                   n             y  ...     n         n            y
4    democrat                   y             y  ...     y         y            y

[5 rows x 17 columns]

In [18]:
# replace '?' with Nan, and strings to numbers
df = df.replace({'y': 1,'n': 0,'?':np.NaN})
df.head()

party  handicapped-infants  ...  duty-free  south-africa
0  republican                  0.0  ...        0.0           1.0
1  republican                  0.0  ...        0.0           NaN
2    democrat                  NaN  ...        0.0           0.0
3    democrat                  0.0  ...        0.0           1.0
4    democrat                  1.0  ...        1.0           1.0

[5 rows x 17 columns]

In [19]:
dem = df[df['party'] == 'democrat']
dem.head()

party  handicapped-infants  water-project  ...  crime  duty-free  south-africa
2  democrat                  NaN            1.0  ...    1.0        0.0           0.0
3  democrat                  0.0            1.0  ...    0.0        0.0           1.0
4  democrat                  1.0            1.0  ...    1.0        1.0           1.0
5  democrat                  0.0            1.0  ...    1.0        1.0           1.0
6  democrat                  0.0            1.0  ...    1.0        1.0           1.0

[5 rows x 17 columns]

In [20]:
rep = df[df['party'] == 'republican']
rep.head()

party  handicapped-infants  ...  duty-free  south-africa
0   republican                  0.0  ...        0.0           1.0
1   republican                  0.0  ...        0.0           NaN
7   republican                  0.0  ...        NaN           1.0
8   republican                  0.0  ...        0.0           1.0
10  republican                  0.0  ...        0.0           0.0

[5 rows x 17 columns]

In [21]:
df['immigration'].isnull().sum()

7

In [22]:
df['immigration'].value_counts()

1.0    216
0.0    212
Name: immigration, dtype: int64

# T-tests

In [0]:
from scipy import stats
import numpy as np

In [0]:
def t

# el salvador aid

## test 1

1. null hypothesis: republican support is split for 'el salvador aid' bill

2. alternate hypothesis: republican support is not 50/50 or .5

3. confidence level: 95%

In [70]:
stats.ttest_1samp(rep['el-salvador-aid'], .5, nan_policy='omit')

Ttest_1sampResult(statistic=26.920515819250607, pvalue=4.531756691646779e-62)

Conclusion:

based on t-statistic of 26.9205 and p-value of ~0, I reject the null hypothesis that republican support the el salvador aid bill is 50/50. I suggest the alternative that levels of support for this bill are different.

## test 2

1. republican support is 95/5
2. republican support is not 95/5 split
3. 95%

In [67]:
stats.ttest_1samp(rep['el-salvador-aid'], .95, nan_policy='omit')

Ttest_1sampResult(statistic=0.09033730140688512, pvalue=0.9281294156190714)

conclusion:

based on t-statistic of 0.6865 and pvalue of 0.4933, I fail to reject the null hypothesis that republican support is split 95/5

# budget


## test 1

1. null: democrat support is split to for budget bill
2. alternate: democrat support is not 50/50
3. confidence: 95%

In [58]:
stats.ttest_1samp(dem['budget'], .5, nan_policy='omit')

Ttest_1sampResult(statistic=19.859406568628835, pvalue=5.75931504660857e-54)

conclusion:

based on t-statistic of 19.8594 and pvalue of ~0, I reject the null hypothesis that democrats support the 'budget' bill is 50/50. I suggest the alternative that democrats do not support the bill 50/50.

## test 2

1. democrat support is split at 90/10 for budget bill
2. democrat support is not 90/10
3. confidence: 95%

In [68]:
stats.ttest_1samp(dem['budget'], .9, nan_policy='omit')

Ttest_1sampResult(statistic=-0.5898833634246224, pvalue=0.5557831066865565)

conclusion:

based on t-statistic of -0.5898 and pvalue of 0.5557, I fail to reject the null hypothesis that democrat support is split at 90/10 for the budget bill

# immigration
1. Null Hypothesis: democrats support for the immigration bill is equal to republican support for same bill 
2. alternative hypothesis: democrat support is not equal to republican support for immigration bill
3. confidence level: 95% confidence, 5% critical value

In [32]:
rep['immigration'].value_counts()

1.0    92
0.0    73
Name: immigration, dtype: int64

In [33]:
dem['immigration'].value_counts()

0.0    139
1.0    124
Name: immigration, dtype: int64

In [38]:
stats.ttest_1samp(dem['immigration'], .5,nan_policy='omit')

Ttest_1sampResult(statistic=-0.9246846884388146, pvalue=0.35598081794452796)

In [39]:
stats.ttest_1samp(rep['immigration'], .5,nan_policy='omit')

Ttest_1sampResult(statistic=1.4845341263724807, pvalue=0.1395867786115413)

In [31]:
# hypothesis testing - an issue democrats support more than republicans
stats.ttest_ind(dem['immigration'],rep['immigration'], nan_policy='omit')

Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)

In [42]:
0.08330248490425066 > 0.01

True

T statistics: -1.7359

P-value: 0.0833

Conclusions:

based on t-statistics of -1.7359 and pvalue of 0.0833, I failed to reject the null hypothesis that democrat support and republican support for the immigration bill is at same level/rate.